RFC/src/RFC-0151_TransactionProtocol.md

```diff
@@ -0,0 +1,255 @@
+  party's satisfaction with the terms.
+
+### The issue with multiple recipients
+
+Each party involved in a Tari transaction must produce a partial signature signing the same challenge. This challenge is
+defined as
+
+$$ e = H(\Sigma R_i \Vert \Sigma P_i \Vert m) $$
+
+where \\(R_i\\) are public nonces generated by each party for this signature; \\(P_i\\) are the public [Spending Key]s;
+and _m_ is the additional metadata for the transaction. \\(\Sigma P_i\\) is the value of the (pre-offset) excess that is
+stored in the transaction kernel.
+
+Notice that every signing party needs to know the sum of all the nonces and public spending keys. This suggests that
+every party knows how many parties are involved in the transaction, which is not an ideal privacy scenario. It would be
+preferable if a secure scheme could be found where each recipient interacts only with the sender and does not need to
+calculate these sums themselves.
+
+Unfortunately, as we discuss below, it seems that it's not possible using any known scheme that satisfies both this
+privacy goal while achieving the desired privacy level.
+
+
+### The issue with multiple senders
+
+To increase privacy, the public excess values are [offset] by a constant random value. The choice of this value, as well
+as fee selection can only be set once per transaction. The privilege of selecting these values is generally bestowed on
+the sender, since she pays the fee. Allowing multiple sending parties (or equivalently, allowing recipients to provide
+inputs) would require a negotiation round to set the fee and offset before the transaction could be constructed. This is
+a complication we don't want to deal with, and so all schemes presented here allow exactly one sender.
+
+### Two-party transactions
+
+Two party transactions are fairly straightforward and are described in detail on TLU. (See [Mimblewimble transaction]).
+
+It is proposed that Tari implement this single-round 2-party transaction scheme as a special case to support both online
+2-party transactions as well as "offline" transactions such as via e-mail, text-message, carrier pigeon etc.
+
+
+###  Multiple recipient transaction scheme
+
+<div class="mermaid">
+sequenceDiagram
+    participant Sender
+    participant Receivers
+# Sender Initializes the transaction
+    activate Sender
+        Sender-->>Sender: initialize
+    deactivate Sender
+# Sender transmits initial data to all receivers
+    activate Sender
+        Sender-->>+Receivers: [tx_id, amt_i,H(Rs||Xs),m]
+        note left of Sender: CollectingCommitments
+        note right of Receivers: SendingCommitment
+        Receivers-->>-Sender: [tx_id, H(Ri||Pi)]
+     deactivate Sender
+# Receiving Outputs
+     activate Sender
+        Sender-->>+Receivers: [tx_id, [Hashed commitments]]
+        note left of Sender: CollectingPubKeys
+        note right of Receivers: SendingOutput
+        Receivers-->>-Sender: [tx_id, Pi, Ri, C_i, RP_i]
```

Comment:
I am not certain what C_i and RP_i are in this step. Guessing they are analogues of terms from MuSig but it is not clear from this context.